ArmRobot/Assets/Scripts/RandomizerPositionObject.cs

```diff
@@ -0,0 +1,165 @@
+            float randomSignX = Random.value;
+            float criticalValueForSign = 0.5f;
+            if (randomSignX > criticalValueForSign){
+                randomX = minRadius + (maxRadius - minRadius) * Random.value;
+            }
+            else {
+                randomX = -maxRadius + (-minRadius - (-maxRadius)) * Random.value;
+            }
+
+            float randomSignZ = Random.value;
+            if (randomSignZ > criticalValueForSign){
+                randomZ = minRadius + (maxRadius - minRadius) * Random.value;
+            }
+            else {
+                randomZ = -maxRadius + (-minRadius - (-maxRadius)) * Random.value;
+            }
+            Vector2 randomPoint = new Vector2(randomX, randomZ); // now we are sure the point is in the area reachable by the robot 
+
+            // now we need to check if their is no conflict with the position of objects already moved 
+            // we define a flag that will give information on whether or not the random point fits 
+            bool flag = true; 
+            
+            // we iterate through all the list of already moved objects 
+            for (var i = 0; i < listOfAlreadyMovedObjects.Count; i++){
+                gameobjectSeen = listOfAlreadyMovedObjects[i];
+
+                // first we will compute the distance between the datapoint generated and one of the object of the list of already moved objects 
+                Vector2 inter_object = new Vector2(gameobjectSeen.transform.position.x - randomX, gameobjectSeen.transform.position.z - randomZ);
+                float distance_between_objects = inter_object.magnitude;
+                
+                // now we need to access to the radius of the gameobjectSeen
+                float gameobjectSeenRadius = 0f; 
+                if (gameobjectSeen.tag == "Sphere"){
+                    gameobjectSeenRadius = gameobjectSeen.GetComponent<SphereCollider>().radius * gameobjectSeen.GetComponent<Transform>().localScale[0]; // for the sphere 
+                }
+                else if (gameobjectSeen.name == "Cube"){
+                    gameobjectSeenRadius = Mathf.Sqrt(2) * gameobjectSeen.GetComponent<BoxCollider>().bounds.extents.x;
+                }
+                else{
+                    gameobjectSeenRadius = gameobjectSeen.GetComponent<CapsuleCollider>().radius * gameobjectSeen.GetComponent<Transform>().localScale[0]; // for the cylinder
+                }
+
+                // now we need to access to the radius of the current object
+                float objectRadius = 0f;
+                if (tag == "Sphere"){
+                    objectRadius = GetComponent<SphereCollider>().radius * GetComponent<Transform>().localScale[0]; // for the sphere 
+                }
+                // otherwise it is a cylinder 
+                else{
+                    objectRadius = GetComponent<CapsuleCollider>().radius * GetComponent<Transform>().localScale[0]; 
+                }
```

Comment:
If you make a `GameObjectRadius(GameObject gameObject){}` method as suggested above, this whole section could be just one line.